polaris/dataset/_dataset.py

```diff
@@ -154,9 +154,9 @@ def _validate_model(cls, m: "Dataset"):
 
         return m
 
-    @field_validator("default_adapters")
+    @field_validator("default_adapters", mode="before")
```

Comment:
Nice usage of `before` here! We should probably use that in other places too so we can simplify the type definitions of our Pydantic models.